### ISSUE TITLE
ObjectScope .Container not working when using property injection

### ISSUE THREAD

**Issue by marcorei** [Swinject/Swinject#133]
The following code will instantiate `A` twice even though it's supposed to be a Singleton.
```Swift
class A {
    init(b: B) {
        print("class A instatiated")
    }
}

class B {
    weak var a: A?
    init() {
        print("class B instatiated")
    }
}

let container = Container()

container.register(A.self) {
    r in
    A(b: r.resolve(B.self)!)
}.inObjectScope(.Container)

container.register(B.self) {
    r in
    B()
}.initCompleted {
    r, t in
    t.a = r.resolve(A.self)!
}.inObjectScope(.Container)

container.resolve(A.self)!
container.resolve(B.self)!
```
Prints:
```
class B instatiated
class A instatiated
class A instatiated
```

**Comment by yoichitgy** [comment]
Fixed on `v1` branch. Still need to merge it into `master` branch.

**Comment by yoichitgy** [comment]
Merged to master! Version 1.1.3 and 2.0.0 beta 2 will have the fix. Thanks @marcorei👍
